app/src/main/res/layout/view_pin_list_item.xml

```diff
@@ -17,6 +17,27 @@
         android:contentDescription="@string/pin_thumbnail_text"
         android:src="@drawable/ic_camera" />
 
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_gravity="top"
+        android:layout_margin="4dp">
+
+        <com.google.android.material.chip.Chip
+            android:id="@+id/pinlist_pin_source"
+            style="@style/TextAppearance.Pinpoint.ImageOverlay"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:clickable="false"
+            android:focusable="false"
+            app:chipEndPadding="0dp"
+            app:chipMinHeight="20dp"
+            app:chipMinTouchTargetSize="0dp"
+            app:chipStartPadding="0dp"
```

Comment:
minor nitpick, looks like there's somehow a little padding at the start of the text. (The chip text looks a tiny bit off center). If you can find a way to improve that, it would be perfect